src/virtual_dom/listener.rs

```diff
@@ -37,6 +82,21 @@ pub struct Listener<Ms> {
     message: Option<Ms>,
 }
 
+impl<Ms: Clone> Clone for Listener<Ms> {
+    fn clone(&self) -> Self {
+        Self {
+            trigger: self.trigger,
+            handler: self.handler.clone(),
+            // closure shouldn't be cloned since the new Listener isn't related to this Listener
+            closure: None,
```

Comment:
Could you explain it a little bit more, please? I don't fully understand it, thanks.

Reply:
> So the problem is that Listener<Ms> doesn't implement Clone, it also contains EventHandler<Ms> and Closure<_> both don't implement Clone too.
> 
> First let's talk about EventHandler<Ms> which contains a closure Box<dyn FnMut(web_sys::Event) -> Ms>, this closure will be used to create Closure<_> in attach function.
> 
> Then we have Closure<_> which get created in attach function, and passed to add_event_listener_with_callback which add event listener in js world, Listener<Ms> stores this closure so it can detach it later with detach function.
> 
> If I understood correctly, EventHandler<Ms> must be cloned when we clone Listener<Ms>, while Closure<_> shouldn't be cloned when we clone Listener<Ms>.
> 
> So we have to implement Clone for EventHandler<Ms>, one way to do that is by wrap type that contain Rc. Then implement Clone for Listener<Ms>.
> 
> Please correct me if I miss something, if I my understood is correct I can open PR for this.

Reply:
I think you know about it more than me now and it make sense. And it looks like another VDOM part which I want to update. Let's leave it as is for now.